### PR TITLE
Add support for bastion hosts and compute routers & nats

### PR DESCRIPTION
**Pull request by umairidris**
Also add new helper funcmap `hcl`

### CODE REVIEW

File: examples/tfengine/generated/full/live/prod/team1/example-prod-apps/project/terragrunt.hcl [hunk 1]
```diff
@@ -29,5 +30,6 @@ dependency "networks" {
 }
 
 inputs = {
-  folder_id = dependency.parent_folder.outputs.name
+  folder_id = "${dependency.parent_folder.outputs.name}"
```
Comment (MartinPetkov) [inline examples/tfengine/generated/full/live/prod/team1/example-prod-apps/project/terragrunt.hcl:33]: Is this correct? I've previously seen it be like it was before, without the `${}`.

File: templates/tfengine/components/project/bastion_hosts/main.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */ -}}
+
+{{range .bastion_hosts -}}
+module "{{resourceName .name}}" {
+  source = "git::https://github.com/terraform-google-modules/terraform-google-bastion-host.git?ref=umairidris-patch-1"
```
Comment (MartinPetkov) [inline templates/tfengine/components/project/bastion_hosts/main.tf:17]: Leave a TODO this this should change to an official release, once available.

File: examples/tfengine/generated/full/live/prod/team1/example-prod-apps/project/terragrunt.hcl (hunk 1 above)
Comment (umairidris) [reply examples/tfengine/generated/full/live/prod/team1/example-prod-apps/project/terragrunt.hcl:33]: I changed the template to use the new `hcl` func in funcmap, that's why this is now needed. Before its value was just set directly.

Comment (MartinPetkov) [reply examples/tfengine/generated/full/live/prod/team1/example-prod-apps/project/terragrunt.hcl:33]: I see. How come terragrunt can read this kind of syntax though, in the generated files?

Comment (umairidris) [reply examples/tfengine/generated/full/live/prod/team1/example-prod-apps/project/terragrunt.hcl:33]: They added support for it.

https://github.com/gruntwork-io/terragrunt/blob/master/config/config.go#L58
https://github.com/gruntwork-io/terragrunt/blob/master/config/config.go#L349

File: templates/tfengine/components/project/bastion_hosts/main.tf (hunk 2 above)
Comment (umairidris) [reply templates/tfengine/components/project/bastion_hosts/main.tf:20]: Done.

File: examples/tfengine/generated/full/live/prod/team1/example-prod-apps/project/terragrunt.hcl (hunk 1 above)
Comment (MartinPetkov) [reply examples/tfengine/generated/full/live/prod/team1/example-prod-apps/project/terragrunt.hcl:33]: I see.

File: examples/tfengine/full.hcl [hunk 3]
```diff
@@ -281,7 +334,8 @@ data "google_secret_manager_secret_version" "db_password" {
   project = "example-secrets"
 }
 EOF
-      } */
+ */
```
Comment (xingao267) [inline examples/tfengine/full.hcl:337]: make this align with the /* TODO(user) ?

Comment (umairidris) [reply examples/tfengine/full.hcl:337]: This is done by the terragrunt hclfmt

**Pull request merged**